### PR TITLE
replacing execv in exec.rs with std::process::Command

### PR DESCRIPTION
**Pull request by rrxtns**
Many many thanks to @smurawski for helping me make this happen.  He's pretty awesome!

This addresses issue: https://github.com/habitat-sh/habitat/issues/1285

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/hab/src/exec.rs [hunk 1]
```diff
@@ -33,36 +31,23 @@ use error::{Error, Result};
...
-/// Note that if successful, this function will not return.
+/// Makes an `std::process::Command` call.
 ///
 /// # Failures
 ///
 /// * Command and/or command arguments cannot be converted into `CString`
 pub fn exec_command(command: PathBuf, args: Vec<OsString>) -> Result<()> {
-    // A massive thanks to the `exec` crate which pointed to the correct invocation
-    // behavior--namely to pass null-terminated string pointers.
     //
```
Comment (smurawski) [inline components/hab/src/exec.rs]: We don't need this extra comment line.

File: components/hab/src/exec.rs [hunk 2]
```diff
@@ -33,36 +31,23 @@ use error::{Error, Result};
...
-    let arg_cstrings = try!(args.into_iter()
-        .map(|arg| CString::new(arg.as_os_str().as_bytes()))
-        .collect::<std::result::Result<Vec<_>, _>>());
-    let mut arg_charptrs: Vec<_> = arg_cstrings.iter()
-        .map(|arg| arg.as_bytes_with_nul().as_ptr() as *const i8)
-        .collect();
-    arg_charptrs.insert(0,
-                        prog_cstring.clone().as_bytes_with_nul().as_ptr() as *const i8);
-    arg_charptrs.push(ptr::null());
+    //
```
Comment (smurawski) [inline components/hab/src/exec.rs]: same here

**Comment by rrxtns** [comment]
I built hab, binlinked it, then tested as follows:

```
[36][default:/src/habitat/components/hab:0]# /bin/hab --version
hab 0.11.0-dev/20161003180925
[38][default:/src/habitat/components/hab:1]# /bin/hab pkg exec core/bash bash -c "exit 0"
[39][default:/src/habitat/components/hab:0]#
[39][default:/src/habitat/components/hab:0]# /bin/hab pkg exec core/bash bash -c "exit 100"
[40][default:/src/habitat/components/hab:100]#
```

**Comment by smurawski** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/165009094).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/165009094).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**